src/site/stages/build/drupal/menus.js

```diff
@@ -0,0 +1,332 @@
+ *
+ */
+function sortMenuLinksWithDepth(menuLinks) {
+  const sortedLinks = []; // The final array of links ordered hierarchically.
+  const roots = []; // This will hold links that are parents of others.
+  const parentChildrenMap = {}; // Describes relationship of parent links to child links.
+  const menuLinkCt = menuLinks.length;
+  let i = 0;
+
+  for (const link of menuLinks) {
+    // Add in a children property so we can have a hierachy.
+    link.children = [];
+
+    // Collect the root items. Roots do not have parents.
+    if (link.parent === null) {
+      roots.push(link);
+
+      // All other links are children and need to be cached in an array
+      // that is keyed by the parent.
+      // Menu items always have a parent or null (it can't be undefined).
+    } else if (link.parent) {
+      // Drupal uses this pattern for the parent property:
+      //   menu-link-content:[the-parent-uuid]
+      // So, we need to split that up to get the parent's uuid.
+      const rootUuid = link.parent.split(':')[1];
+      if (!parentChildrenMap[rootUuid]) {
+        parentChildrenMap[rootUuid] = [];
+      }
+      parentChildrenMap[rootUuid].push(link);
+    }
+  }
+
+  // We go through each parent and push its children into a 'children' array.
+  while (roots.length > 0) {
+    // We use an index i to ensure that we do not exceed our 'backstop:'
+    // the total number of links in the menu. Thus, we prevent an infinite loop
+    // and stop the build with an error.
+    if (i >= menuLinkCt) {
+      throw new Error('Drupal header menu data contained an anomaly.');
+    }
+
+    // Grab first item from roots array.
+    const root = roots.shift();
+
+    // If this item is in the top level of the menu (no parent),
+    // push it into our sorted list.
+    if (root.parent === null) {
+      sortedLinks.push(root);
+    }
+
+    // Grab the kids from our parentChildrenMap array.
+    if (parentChildrenMap[root.uuid]) {
+      parentChildrenMap[root.uuid].forEach(child => {
+        root.children.push(child);
+
+        // Put this item into the first position of the parents array
+        // so we can find its children, if needed.
+        // Note that this is how we are able to recurse to an arbitrary depth of menu.
+        roots.unshift(child);
+      });
+    }
+    i++;
+  }
+
+  return sortedLinks;
+}
+
+function createLinkObj(hostUrl, link) {
+  return {
+    text: link.title,
+    href: convertLinkToAbsolute(hostUrl, link.link.url.path),
+  };
+}
+
+function makeLinkList(hostUrl, links) {
+  const list = [];
+
+  links.forEach(link => {
+    const linkObj = createLinkObj(hostUrl, link);
+    list.push(linkObj);
+  });
+
+  return list;
+}
+
+function makePromo(hostUrl, promo) {
+  const img = promo.entity.fieldImage.entity.image;
+  const link = promo.entity.fieldPromoLink.entity.fieldLink;
+
+  return {
+    img: {
+      src: convertLinkToAbsolute(hostUrl, img.derivative.url),
+      alt: img.alt || '',
```

Comment:
I don't know what the alternative is here, but it's worth raising this to @1Copenut. :man_shrugging:

Reply:
This implementation should keep us in axe's good graces. The image will either return it's alt text or an empty string. Both are valid in the spec.

These images are largely decorative, given we have a descriptive header associated with the links beneath them, so an `alt=""` is a good thing here. Screen readers will ignore the image as decorative, not adding any contextual substance to the page.

If an image *does* have alt text, it's a small amount of noise, but not a deal breaker.